### PR TITLE
[API] Search products by name within specific taxon

### PR DESCRIPTION
**Pull request by pamil**

### CODE REVIEW

**Review by Tomanhez — APPROVED** [review]

File: src/Sylius/Behat/Client/ApiClientInterface.php [hunk 1]
```diff
@@ -60,6 +60,9 @@ public function addParameter(string $key, $value): void;
     /** @param string|int $value */
     public function addFilter(string $key, $value): void;
 
+    /** @param string|int $value */
```
Comment (AdamKasp) [inline src/Sylius/Behat/Client/ApiClientInterface.php:63]:
```suggestion
```

**Review by AdamKasp — APPROVED** [review]

File: src/Sylius/Behat/Client/ApiPlatformClient.php [hunk 2]
```diff
@@ -211,6 +211,12 @@ public function addFilter(string $key, $value): void
         $this->addParameter($key, $value);
     }
 
+    /** @param string|int $value */
```
Comment (AdamKasp) [inline src/Sylius/Behat/Client/ApiPlatformClient.php:214]:
```suggestion
```

File: src/Sylius/Behat/Context/Api/Shop/ProductContext.php [hunk 3]
```diff
@@ -65,6 +65,46 @@ public function iBrowseProductsFromTaxon(TaxonInterface $taxon): void
...
+    public function iClearFilter(): void
+    {
+        $this->client->clearParameters();
+        $this->client->filter();
+    }
+
+    /**
+     * @When I search for products with name :name
+     */
+    public function iSearchForProductsWithName(string $name)
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Shop/ProductContext.php:80]: Missing return type

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Kamil! :1st_place_medal:

**Pull request merged**